### PR TITLE
(DOCS-4353) Add page for creating an agent-integration for marketplace

### PR DESCRIPTION
**Pull request by hestonhoffman**
- Adds new page for creating an agent-integration for marketplace.
- Shares readme and media carousel sections with offering page
- Updates the ddev installation instructions
---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/developers/marketplace/marketplace-agent-integration.md [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Open a pull request
+
+Before you open a pull request, run the following command to catch any problems with your integration:
+
+```
+ddev validate all changes
```
Comment (hestonhoffman) [inline content/en/developers/marketplace/marketplace-agent-integration.md:475]:
```suggestion
ddev validate all <INTEGRATION_NAME>
```

File: content/en/developers/marketplace/marketplace-agent-integration.md [hunk 2]
```diff
@@ -0,0 +1,442 @@
+---
+title: Create an Agent-based Integration for Marketplace
+type: documentation
+description: Learn how to create an Agent-based integration for the marketplace repo.
```
Comment (alai97) [inline content/en/developers/marketplace/marketplace-agent-integration.md:4]:
```suggestion
description: Learn how to create an Agent-based integration for the Datadog Marketplace repository.
```

File: content/en/developers/marketplace/marketplace-agent-integration.md [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+   ```
+
+2. Run the `ddev` command with the `-t tile` option:
+   ```
+   ddev create "<INTEGRATION NAME>"
+   ```
+
+## Write an Agent Check
+
+At the core of each Agent-based integration is an *Agent Check* that periodically collects information and sends it to Datadog. Checks inherit their logic from the `AgentCheck` base class and have the the following requirements:
```
Comment (alai97) [inline content/en/developers/marketplace/marketplace-agent-integration.md:98]:
```suggestion
At the core of each Agent-based integration is an *Agent Check* that periodically collects information and sends it to Datadog. Checks inherit their logic from the `AgentCheck` base class and have the following requirements:
```

File: content/en/developers/marketplace/marketplace-agent-integration.md [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+
+- Integrations running on the Datadog Agent v7 and later must be compatible with Python 3; however, Agents v5 and v6 still use Python 2.7.
```
Comment (alai97) [inline content/en/developers/marketplace/marketplace-agent-integration.md:100]:
```suggestion
- Integrations running on the Datadog Agent v7 or later must be compatible with Python 3; however, Agents v5 and v6 still use Python 2.7.
```

File: content/en/developers/marketplace/marketplace-agent-integration.md [hunk 5]
```diff
@@ -0,0 +1,442 @@
...
+
+### Write an integration test
+
+The [unit test above](#write-a-unit-test) doesn't check the collection logic. To test the logic, you need to create an environment for an integration test and write an integration test.
+
+#### Create an environment for the integration test
+
+To create an environment for your integration test, create a Docker Compose file at `INTEGRATION_NAME/tests/docker-compose.yml`
+
+For the example check, the toolkit uses `docker` to spin up an Nginx container and allows the check to retrieve the welcome page. The `docker-compose.yml` file has the following contents:
```
Comment (alai97) [inline content/en/developers/marketplace/marketplace-agent-integration.md:217]:
```suggestion
For the example check, the toolkit uses `docker` to spin up an NGINX container and allows the check to retrieve the welcome page. The `docker-compose.yml` file has the following contents:
```

File: content/en/developers/marketplace/marketplace-agent-integration.md [hunk 6]
```diff
@@ -0,0 +1,442 @@
...
+
+
+@pytest.fixture
+def instance():
+    return INSTANCE.copy()
+{{< /code-block >}}
+
+#### Add an integration test
+
+After you've setup an environment for the integration test, add an integration test to the `INTEGRATION_NAME/tests/test_INTEGRATION_NAME.py` file:
```
Comment (alai97) [inline content/en/developers/marketplace/marketplace-agent-integration.md:266]:
```suggestion
After you've set up an environment for the integration test, add an integration test to the `INTEGRATION_NAME/tests/test_INTEGRATION_NAME.py` file:
```

File: content/en/developers/marketplace/marketplace-agent-integration.md [hunk 7]
```diff
@@ -0,0 +1,442 @@
...
+
+{{% integrations_readme %}}
+
+### Media carousel
+
+{{% integrations_media_carousel %}}
+
+## Build the wheel
+
+The `pyproject.toml` file provides the metadata that is used to package and build the wheel. The wheel contains the files necessary for the functioning of the integration itself, which includes the Check, configuration example file, and artifacts generated during the build of the wheel.
```
Comment (alai97) [inline content/en/developers/marketplace/marketplace-agent-integration.md:347]: "Functioning of the integration" sounded off to me, but feel free to reword or ignore!

```suggestion
The `pyproject.toml` file provides the metadata that is used to package and build the wheel. The wheel contains the files necessary for the integration to function itself, and includes the Check, configuration example file, and artifacts generated during the build of the wheel.
```

File: content/en/developers/marketplace/marketplace-agent-integration.md [hunk 8]
```diff
@@ -0,0 +1,442 @@
...
+All additional elements, including the metadata files, are not meant to be contained within the wheel, and are used elsewhere by the Datadog platform and ecosystem. To learn more about Python packaging, see [Packaging Python Projects][20].
+
+Once your `pyproject.toml` is ready, create a wheel:
+
+- (Recommended) With the `ddev` tooling: `ddev release build <INTEGRATION_NAME>`.
+- Without the `ddev` tooling: `cd <INTEGRATION_DIR> && pip wheel . --no-deps --wheel-dir dist`.
+
+## Install the wheel
+
+The wheel is installed using the Agent `integration` command, available in [Agent v6.10.0 and up][21]. Depending on your environment, you may need to execute this command as a specific user or with specific privileges:
```
Comment (alai97) [inline content/en/developers/marketplace/marketplace-agent-integration.md:358]:
```suggestion
The wheel is installed using the Agent `integration` command, available in [Agent v6.10.0 or later][21]. Depending on your environment, you may need to execute this command as a specific user or with specific privileges:
```

File: content/en/developers/marketplace/marketplace-agent-integration.md [hunk 9]
```diff
@@ -0,0 +1,442 @@
...
+
+```
+ddev validate all <INTEGRATION_NAME>
+```
+
+Push up your feature branch and open a pull request that contains your integration tile's asset files (including images) in the [`marketplace` repository][7]. The Marketplace repository does not allow forks. For instructions on creating a clone of the repo, see the [Set up section](#set-up-a-directory-and-clone-the-marketplace-repository). After you've created your pull request, automatic checks run in Azure DevOps pipelines to verify that your pull request is in good shape and contains all the required content to be updated.
+
+To request access to the Azure DevOps pipeline, leave a comment in the pull request requesting access.
+
+After you've created your pull request, automatic checks run to verify that your pull request is in good shape and contains all the required content to be updated.
```
Comment (alai97) [inline content/en/developers/marketplace/marketplace-agent-integration.md:401]: Extra sentence here I think

```suggestion
```

File: content/en/developers/marketplace/offering.md [hunk 10]
```diff
@@ -118,47 +112,29 @@ Once you've decided on an offering, set up a directory:
 
 ## Install and configure the Datadog development toolkit
 
-The Datadog Development Toolkit command (`ddev`) allows you to create scaffolding when you are developing an integration by generating a skeleton of your integration tile's assets and metadata.
+The Agent Integration Developer Tool allows you to create scaffolding when you are developing an integration by generating a skeleton of your integration tile's assets and metadata. For instructions on installing the tool, seeInstall the [Datadog Agent Integration Developer Tool][17].
```
Comment (alai97) [inline content/en/developers/marketplace/offering.md:115]:
```suggestion
The Agent Integration Developer Tool allows you to create scaffolding when you are developing an integration by generating a skeleton of your integration tile's assets and metadata. For instructions on installing the tool, see [Install the Datadog Agent Integration Developer Tool][17].
```

File: content/en/developers/marketplace/offering.md [hunk 11]
```diff
@@ -118,47 +112,29 @@ Once you've decided on an offering, set up a directory:
...
 
-Before you begin, make sure you meet the following prerequisites:
+Once you've installed the Developer tool, configure the tool for the `marketplace` repo:
```
Comment (alai97) [inline content/en/developers/marketplace/offering.md:117]:
```suggestion
Once you've installed the Developer tool, configure the tool for the `marketplace` repo.
```

File: content/en/developers/marketplace/offering.md [hunk 12]
```diff
@@ -118,47 +112,29 @@ Once you've decided on an offering, set up a directory:
...
+ddev config set marketplace <PATH/TO/MARKETPLACE>
+ddev config set repo marketplace
+```
 
 ## Populate the integration tile scaffolding
 
-Run the `ddev` command to generate a skeleton of the folders and files needed for your integration. The options you use with the command are different depending on what type of integration you are developing. For a full list of the files created by the `ddev` command, see [Integrations assets][22].
-
-### Create an informational tile only listing
+Run the `ddev` command to generate a skeleton of the folders and files needed for your integration. The options you use with the command are different depending on what type of integration you are developing. For a full list of the files created by the `ddev` command, see [Integrations assets][19].
```
Comment (alai97) [inline content/en/developers/marketplace/offering.md:137]:
```suggestion
Run the `ddev` command to generate a skeleton of the folders and files needed for your integration. The options you use with the command are different depending on what type of integration you are developing. For a full list of the files created by the `ddev` command, see the [Integrations assets](#complete-the-necessary-integration-asset-files) section below.
```

**Review by alai97 — CHANGES_REQUESTED** [review]
Some copy suggestions!

**Review by alai97 — APPROVED** [review]
Looks great, amazing work!

**Pull request closed without merging**